pkg/model/openstackmodel/firewall.go

```diff
@@ -465,15 +474,90 @@ func (b *FirewallModelBuilder) addProtokubeRules(c *fi.ModelBuilderContext, sgMa
 				PortRangeMin: i(portRange.Min),
 				PortRangeMax: i(portRange.Max),
 			}
-			addDirectionalGroupRule(c, masterSG, nodeSG, protokubeRule)
-			addDirectionalGroupRule(c, nodeSG, masterSG, protokubeRule)
-			addDirectionalGroupRule(c, masterSG, masterSG, protokubeRule)
-			addDirectionalGroupRule(c, nodeSG, nodeSG, protokubeRule)
+			b.addDirectionalGroupRule(c, masterSG, nodeSG, protokubeRule)
+			b.addDirectionalGroupRule(c, nodeSG, masterSG, protokubeRule)
+			b.addDirectionalGroupRule(c, masterSG, masterSG, protokubeRule)
+			b.addDirectionalGroupRule(c, nodeSG, nodeSG, protokubeRule)
 		}
 	}
 	return nil
 }
 
+func (b *FirewallModelBuilder) getExistingRules(sgMap map[string]*openstacktasks.SecurityGroup) error {
+	osCloud, err := b.createCloud()
+	if err != nil {
+		return err
+	}
+	sgIdMap := make(map[string]*openstacktasks.SecurityGroup)
+	for sgName, sgt := range sgMap {
+
+		sgs, err := osCloud.ListSecurityGroups(sg.ListOpts{
+			Name: sgName,
+		})
+		if err != nil {
+			return err
+		}
+		if len(sgs) != 1 {
+			continue
```

Comment:
if we have two sgs with same name? in that case imo we should print log about that

Reply:
Erroring out now if that is the case. Thanks for catching.